File: php/src/Model/ResultCodes.php
<?php

declare(strict_types=1);

namespace SlaunchX\Plutus\Model;

/**
 * 统一响应包络中 `code` 字段的成功码集合。
 *
 * **本类仅供文档与便利用途, 不参与成功判定。** 成功与否的唯一权威是包络里的
 * `success` 布尔字段 (见 {@see ApiResponse::isSuccess()}): 该字段存在且为 JSON 布尔时
 * 直接采用其值, 否则回退到 HTTP 2xx。请勿用 `isSuccessCode()` 代替 `isSuccess()`。
 *
 * 失败时 `code` 为字符串错误码: 网关层是 `域.名称` (如 `API.SIGNATURE_INVALID`),
 * 业务层是 4xxx/5xxx 的数字字符串 (如 `"4022"`)。
 */
final class ResultCodes
{
    /**
     * 成功族码。`"2101"` 表示账号待审批 —— 登录成功但不签发 JWT, `success` 仍为 `true`,
     * 因此它是成功码, 不是错误码。
     *
     * @var array<int, string>
     */
    public const SUCCESS_CODES = ['2000', '2001', '2002', '2004', '2006', '2101'];

    /** 账号待审批: 成功码, 但调用方需引导用户等待审批。 */
    public const ACCOUNT_PENDING_APPROVAL = '2101';

    /**
     * 该码是否属于成功族。
     *
     * 仅作文档与便利用途; **不是**成功判定依据, 判定见 {@see ApiResponse::isSuccess()}。
     */
    public static function isSuccessCode(string $code): bool
    {
        return in_array($code, self::SUCCESS_CODES, true);
    }
}

File: go/result_codes.go
package plutus

// 统一响应包络的成功族结果码。
//
// 这些常量只作文档与便利用途, 不参与成功判定: 成功与否以包络的 success 布尔字段为准
// (缺失时回退 HTTP 状态码), 见 APIResponse.IsSuccess。
const (
	// CodeResultSuccess 是通用成功码。
	CodeResultSuccess = "2000"
	// CodeResultCreated 是资源创建成功码。
	CodeResultCreated = "2001"
	// CodeResultAccepted 是请求已受理成功码。
	CodeResultAccepted = "2002"
	// CodeResultNoContent 是无内容成功码。
	CodeResultNoContent = "2004"
	// CodeResultPartialContent 是部分内容成功码。
	CodeResultPartialContent = "2006"
	// CodeAccountPendingApproval 表示账号待审批: 登录成功但不签发 JWT。
	// 这是一个成功码, 包络的 success 仍为 true。
	CodeAccountPendingApproval = "2101"
)

// SuccessCodes 是统一响应包络的成功码集合。
//
// 仅作文档与便利用途, 不是成功判定依据: 成功判定见 APIResponse.IsSuccess。
// "2101" (账号待审批) 也属于成功码。切勿修改本切片的内容。
var SuccessCodes = []string{
	CodeResultSuccess,
	CodeResultCreated,
	CodeResultAccepted,
	CodeResultNoContent,
	CodeResultPartialContent,
	CodeAccountPendingApproval,
}

// IsSuccessCode 判断 code 是否属于 SuccessCodes。
//
// 仅作文档与便利用途: 判定一次调用是否成功必须用 APIResponse.IsSuccess,
// 它以包络的 success 布尔字段为权威。
func IsSuccessCode(code string) bool {
	for _, c := range SuccessCodes {
		if c == code {
			return true
		}
	}
	return false
}
